cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/DetailedHealthCheck.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.cloudbreak.common.type;
+
+public class DetailedHealthCheck {
+
+    private final HealthCheckType healthCheckType;
+
+    private final HealthCheckResult healthCheckResult;
```

Comment:
A little confused by what the various fields mean w.r.t a service. Is this representing instances of the service on the specific node?, which service, etc?

Reply:
The equivalent class HealthCheck is quite simple - type and result. This one starts getting confusing when certain fields don't apply across the board.
Maybe separate Host / Service Health check representations and let them have more detailed information in themselves?

Reply:
Ack, a separate class for each type of health check with extra info is a better approach. Will update this patch